apis/keda/v1alpha1/scaledobject_types.go

```diff
@@ -239,13 +239,19 @@ func (so *ScaledObject) IsUsingModifiers() bool {
 
 // getHPAMinReplicas returns MinReplicas based on definition in ScaledObject or default value if not defined
 func (so *ScaledObject) GetHPAMinReplicas() *int32 {
-	if so.Spec.MinReplicaCount != nil && *so.Spec.MinReplicaCount > 0 {
+	if so.Spec.MinReplicaCount != nil {
```

Comment:
this was imho better before, this should be used by the HPA controller which then needs to reimplement the logic of figuring out the 0.

Reply:
> this was imho better before, this should be used by the HPA controller which then needs to reimplement the logic of figuring out the 0.

Originally, there is no defaultMaxReplicas in HPA and I will add defaultMaxReplicas and defaultMinReplicas together this time. I think it is more clear in this way. Although, I give up the original logic here.

Reply:
I think that this is just to prevent something like minReplicas -10, but it's true that it's already covered in explicitly outside this function

Reply:
> I think that this is just to prevent something like minReplicas -10, but it's true that it's already covered in explicitly outside this function

One case, when user want to set minReplicas 10, but mis-input as -10. There no error returned from webhook and user can't find the problem, the real minReplicas is 0, which is not the 10 user want to set. I think validation webhook should not be used to set default value for invalid value.

Reply:
No no, I meant that I see worth not returning a default value on invalid user value and this verification is still done from HPA pov with your other addition
```
if scaledObject.Spec.IdleReplicaCount != nil && *idleReplicas < 0 {
	return fmt.Errorf("IdleReplicaCount=%d must not be negative", *idleReplicas)
}

```